### DISCUSSION TITLE
SystemDecorations.BorderOnly has an artifact in the version 11.0.2

### DISCUSSION THREAD

**Discussion by UralMichael** [AvaloniaUI/Avalonia#16727] [General]
**Describe the bug**
I have an application where the main window is configured to not have the default window bar (`SystemDecorations="BorderOnly"`).

I updated Avalonia packages to the new version 11.0.2.

Now the window has an artifact margin on the top side.

**To Reproduce**
1. Create project from the default template `Avalonia .NET App (Avalonia UI)`
2. Update Avalonia and Avalonia.Desktop packages to the version `11.0.2`
3. Set `SystemDecorations="BorderOnly" ` in the main window
4. Build and start the app

MainWindow.axaml:
```
<Window xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        mc:Ignorable="d" d:DesignWidth="800" d:DesignHeight="450"
        x:Class="AvaloniaApplication1.MainWindow"
		SystemDecorations="BorderOnly"
        Title="AvaloniaApplication1">
    Welcome to Avalonia!
</Window>
```
Project file:
```
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <OutputType>WinExe</OutputType>
    <TargetFramework>net7.0</TargetFramework>
    <Nullable>enable</Nullable>
    <BuiltInComInteropSupport>true</BuiltInComInteropSupport>
    <ApplicationManifest>app.manifest</ApplicationManifest>
    <AvaloniaUseCompiledBindingsByDefault>true</AvaloniaUseCompiledBindingsByDefault>
  </PropertyGroup>


  <ItemGroup>
    <PackageReference Include="Avalonia" Version="11.0.2" />
    <PackageReference Include="Avalonia.Desktop" Version="11.0.2" />
    <PackageReference Include="Avalonia.Themes.Fluent" Version="11.0.0" />
    <PackageReference Include="Avalonia.Fonts.Inter" Version="11.0.0" />
    <!--Condition below is needed to remove Avalonia.Diagnostics package from build output in Release configuration.-->
    <PackageReference Condition="'$(Configuration)' == 'Debug'" Include="Avalonia.Diagnostics" Version="11.0.0" />
  </ItemGroup>
</Project>
```

**Screenshots**

11.0.0:
<img src="https://github.com/AvaloniaUI/Avalonia/assets/47430417/e6ff56df-02a9-4027-a492-5d8c22b7e322" alt="11_0_0" height="250"/>

11.0.2:
<img src="https://github.com/AvaloniaUI/Avalonia/assets/47430417/8d45a89a-d352-4b18-b376-81b0abb296a5" alt="11_0_2" height="250"/>

**Desktop:**

- OS: Windows
- Version: 10.0.19044.3086

**Comment by timunie** [comment]
why not updating all packages?

**Comment by UralMichael** [comment]
@timunie
Initially I updated everything to 11.0.2 in my project.
In the template I wanted to find the exact package that causes the artifact.

I've just re-checked the template project with updating everything to 11.0.2 and even to 11.0.3.
The result is the same.

**Comment by kekekeks** [comment]
@emmauss could you check if we have some window styles that shouldn't be there in borderless mode?

**Comment by baochenw** [comment]
Set `ExtendClientAreaTitleBarHeightHint="0"` to workaround it.

**Comment by SCLDGit** [comment]
Can confirm that this is still an issue in 11.1.3. BorderOnly has a small sliver of title bar visible on Windows 11. Additionally, BorderOnly seemingly causes window to not respect CanResize="False"

**Comment by emmauss** [comment]
Forgot to post an explanation here.
On Windows, BorderOnly will for the OS to apply a border on the window. On Windows 10 and newer, that border is invisible on the left, bottom and right of the window, but visible on the top. That's the white line you see in the pics. Its not an artifact, it the OS applied border. This behavior can not be removed for BorderOnly.
As for setting CanResize  being ignored, this should be fixed in master. Due to the behaviors of BorderOnly and CanResize=False being mutually exclusive, CanResize=false essentially removes BorderOnly is enabled, because setting a window Border makes a window resizable
